corebehrt/common/setup.py

```diff
@@ -27,191 +32,360 @@ def get_args(default_config_name, default_run_name=None):
+
+        # Validate and create directories
+        self.check_directory("features")
+        self.check_directory("tokenized")
+        self.check_directory("pretrain_model")
+        self.check_file("outcome")
+        self.check_file("exposure")
+        self.create_run_directory(
+            "model", base="runs", run_id=self.generate_finetune_run_id()
         )
-        if run_folder is None:
-            run_folder = join(cfg.paths.output_path, run_name)
 
-        os.makedirs(run_folder, exist_ok=True)
-        os.makedirs(join(run_folder, CHECKPOINTS_DIR), exist_ok=True)
-        logger = setup_logger(run_folder)
-        logger.info(f"Run folder: {run_folder}")
-        return logger, run_folder
+        # Write config in output directory.
+        self.write_config("model", source="features", name=DATA_CFG)
+        self.write_config("model", source="pretrain_model", name=PRETRAIN_CFG)
+        self.write_config("model", name=FINETUNE_CFG)
+
+        # Add pretrain info to config
+        data_cfg = self.get_config("model", name=DATA_CFG)
+        self.cfg.paths.data = data_cfg.paths.data
+        if "tokenized" not in self.cfg.paths:
+            logger.info("Tokenized dir not in config. Adding from pretrain config.")
+            self.cfg.paths.tokenized = data_cfg.paths.tokenized
+
+    #
+    # Directory naming generators
+    #
+    def generate_run_id(self) -> str:
+        """
+        Generates a run id for naming run folder.
+        If run_name is specified in the paths config, it is returned.
+        """
+        if hasattr(self.cfg.paths, "run_name"):
+            return self.cfg.paths.run_name
 
-    @staticmethod
-    def adjust_paths_for_finetune(cfg: Config) -> Config:
-        """
-        Adjusts the following paths in the configuration for the finetune environment:
-        - output_path: set to pretrain_model_path
-        - run_name: constructed according to setting
-        """
-        pretrain_model_path = cfg.paths.get("pretrain_model_path")
-        model_path = cfg.paths.get("model_path")
-        if model_path is not None:
-            model_path = split(model_path)[
-                0
-            ]  # Use directory of model path (the model path will be constructed in the finetune script)
-        save_folder_path = cfg.paths.get("save_folder_path")
-
-        # Determine the output path with a priority order
-        output_path = pretrain_model_path or model_path or save_folder_path
-        if output_path is None:
-            raise ValueError(
-                "Either pretrain_model_path, model_path, or save_folder_path must be provided."
-            )
-        cfg.paths.output_path = output_path
-        cfg.paths.run_name = DirectoryPreparer.construct_finetune_model_dir_name(cfg)
-        return cfg
+        return uuid.uuid4().hex
 
-    @staticmethod
-    def construct_finetune_model_dir_name(cfg: Config) -> str:
+    def generate_finetune_run_id(self) -> str:
```

Comment:
We could also refactor this function and break down into smaller helper functions.
Also, I think the previous name was more descriptive

Reply:
I'm fine with having it in a single function, as this is _only_ relevant for the naming of the finetune folder. For naming, I can see an argument for both, so don't really care

Reply:
I think I had a reason for using run_id instead of dir_name (maybe azure related), but I do not remember, so it probably ended up not being relevant :-)
Fine to use dir name for me.

Wrt. function splitting - here and in general - I must admit I rarely split (shorter) single-use functions, if I do not think it improves the readability. "shorter" is of course very subjective. Maybe we need a coding guideline on this, so as to not have it as a comment for every PR :-)

Reply:
I renamed it (and renamed `run_id` => `run_name` as well.
I did not break it further down, except the suffix, where we already had another utility function, which did exactly the same.

Reply:
I renamed it (and renamed `run_id` => `run_name` as well).
I did not break it further down, except the suffix, where we already had another utility function, which did exactly the same.